my-fund-backend/src/main/java/com/myfund/controllers/BudgetController.java

```diff
@@ -63,4 +62,46 @@ public ResponseEntity<IncomeDTO> updateIncome(@PathVariable("budgetId") Long bud
         IncomeDTO incomeDTO = budgetService.updateIncome(budgetId, incomeId, createIncomeDTO, user);
         return new ResponseEntity<>(incomeDTO, HttpStatus.OK);
     }
+
+    @GetMapping("/budgets/{budgetId}/categories/{categoryId}/expenses/total")
+    public ResponseEntity<FinancialAggregateCategoryDTO> getTotalExpensesForBudgetAndCategory(@PathVariable("budgetId") Long budgetId, @PathVariable("categoryId") Long categoryId, @AuthenticationPrincipal User user) {
+        FinancialAggregateCategoryDTO totalExpensesByCategory = budgetService.getTotalExpensesByCategory(budgetId, categoryId, user);
+        return new ResponseEntity<>(totalExpensesByCategory, HttpStatus.OK);
+    }
+
+    @GetMapping("/budgets/{budgetId}/subcategories/{subcategoryId}/expenses/total")
+    public ResponseEntity<FinancialAggregateSubcategoryDTO> getTotalExpensesForBudgetAndSubcategory(@PathVariable("budgetId") Long budgetId, @PathVariable("subcategoryId") Long subcategoryId, @AuthenticationPrincipal User user) {
+        FinancialAggregateSubcategoryDTO totalExpensesBySubcategory = budgetService.getTotalExpensesBySubcategory(budgetId, subcategoryId, user);
+        return new ResponseEntity<>(totalExpensesBySubcategory, HttpStatus.OK);
+    }
+
+    @GetMapping("/budgets/{budgetId}/categories/{categoryId}/incomes/total")
+    public ResponseEntity<FinancialAggregateCategoryDTO> getTotalIncomesForBudgetAndCategory(@PathVariable("budgetId") Long budgetId, @PathVariable("categoryId") Long categoryId, @AuthenticationPrincipal User user) {
+        FinancialAggregateCategoryDTO totalIncomesByCategory = budgetService.getTotalIncomesByCategory(budgetId, categoryId, user);
+        return new ResponseEntity<>(totalIncomesByCategory, HttpStatus.OK);
+    }
+
+    @GetMapping("/budgets/{budgetId}/subcategories/{subcategoryId}/incomes/total")
+    public ResponseEntity<FinancialAggregateSubcategoryDTO> getTotalIncomesForBudgetAndSubcategory(@PathVariable("budgetId") Long budgetId, @PathVariable("subcategoryId") Long subcategoryId, @AuthenticationPrincipal User user) {
+        FinancialAggregateSubcategoryDTO totalIncomesBySubcategory = budgetService.getTotalIncomesBySubcategory(budgetId, subcategoryId, user);
+        return new ResponseEntity<>(totalIncomesBySubcategory, HttpStatus.OK);
+    }
+
+    @DeleteMapping("/budgets/{budgetId}")
+    public ResponseEntity<?> deleteBudget(@PathVariable("budgetId") Long budgetId, @AuthenticationPrincipal User user) {
+        budgetService.deleteBudgetByIdAndUser(budgetId, user);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```

Comment:
hmm, for delete i would expect more 200 that 204, 200 means for me success. in dosc 
![image](https://github.com/lukaszkosminski/my-fund/assets/1099625/d20d0cad-fd34-423e-9382-83581b9363b3)
you can read that you should return 202